### PR TITLE
test: monumental source-layer and ORC coverage tranche

### PR DESCRIPTION
**Pull request by ibgreen**
## Goals

- Deliver a materially larger coverage gain than the previous record tranche.
- Target dense, high-impact runtime paths while keeping required test execution fast and hermetic.
- Exercise source-layer lifecycle behavior and protocol edge cases instead of adding shallow import coverage.

## Changes

- Add comprehensive deck.gl source-layer coverage for shared 2D tile traversal, tile lifecycle and rendering, image sources, and raster sources.
- Cover geographic, wrapped-map, Cartesian, transformed, culling, picking, cache, loading, event, error, and cleanup paths.
- Expand ORC coverage across primitive columns, nulls, multi-stripe files, columnar/object-row/empty inputs, validation failures, direct binary values, and dictionary values.
- Correct ORC direct-v2 string/binary encoding metadata and stop treating `DIRECT_V2` as dictionary encoding.
- Expand WMS and WFS coverage across URL normalization, version aliases, axis ordering, metadata, binary features, vendor parameters, abort signals, response validation, and service errors.

## Coverage impact

Compared by exact source statement locations against the merged coverage artifact from `master`:

- **552 previously uncovered statements covered**
- **82.21% → 82.99% projected combined statement coverage**
- **+0.78 percentage points**
- Focused test body: 103 tests in approximately 108 ms

Largest gains:

- Tile2DSourceLayer: 135 statements
- deck tile traversal and adapter: 138 statements
- ORC encoder/parser: 103 statements
- raster and image source layers: 87 statements
- WFS/WMS source runtimes: 84 statements

## Validation

- `yarn lint fix`
- `yarn test-audit`
- `yarn build`
- `yarn build-workers`
- `yarn test-node` — 58 files, 380 passed, 6 skipped
- `yarn test-headless` — 539 files passed, 1 skipped; 3,535 passed, 39 skipped
- Focused Chromium coverage — 7 files, 103 passed

### CODE REVIEW

File: modules/orc/test/orc-loader.spec.ts [hunk 1]
```diff
@@ -284,6 +285,125 @@ test('ORCWriter#encode writes dictionary-encoded repeated binary values', async
...
+  } as any;
+
+  const encoded = encodeORC(table, {orc: {stripeSize: 2}});
+  const result = await ORCLoaderWithParser.parse(encoded);
+  expect(result.shape).toBe('arrow-table');
+  if (result.shape !== 'arrow-table') return;
+  expect(result.data.numRows).toBe(3);
+  expect(result.data.getChild('enabled')?.toArray()).toEqual([true, false, null]);
+  expect(Array.from(result.data.getChild('count')?.toArray() || [])).toEqual([-50_000, 75_000, 0]);
+  expect(result.data.getChild('label')?.toArray()).toEqual(['repeat', 'repeat', null]);
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/orc/test/orc-loader.spec.ts:353]: Fixed in follow-up PR #3911 (commit 02ea2d5bff). FLOAT and DOUBLE data streams now contain raw little-endian IEEE bytes rather than byte-RLE output, and the round-trip test asserts every primitive column including float/double values and nulls. Validated with the full build, worker rebuild, Node suite (380 passing), and Chromium suite (3,535 passing).
